src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationImporter.java

```diff
@@ -29,31 +44,59 @@ public GeneticAlterationImporter(
      * multiple rows for the same gene, and we only want to import the first row.
      */
     public boolean store(
-        String[] values,
-        CanonicalGene gene,
-        String geneSymbol
+            String[] values,
+            CanonicalGene gene,
+            String geneSymbol
     ) throws DaoException {
-        try {
-            if (importSetOfGenes.add(gene.getEntrezGeneId())) {
-                daoGeneticAlteration.addGeneticAlterations(geneticProfileId, gene.getEntrezGeneId(), values);
-                return true;
-            } else {
-                String geneSymbolMessage = "";
-                if (geneSymbol != null && !geneSymbol.equalsIgnoreCase(gene.getHugoGeneSymbolAllCaps())) {
-                    geneSymbolMessage = " (given as alias in your file as: " + geneSymbol + ")";
-                }
-                ProgressMonitor.logWarning(format(
-                    "Gene %s (%d)%s found to be duplicated in your file. Duplicated row will be ignored!",
-                    gene.getHugoGeneSymbolAllCaps(),
-                    gene.getEntrezGeneId(),
-                    geneSymbolMessage)
-                );
-                return false;
-            }
-        } catch (Exception e) {
-            throw new RuntimeException("Aborted: Error found for row starting with " + geneSymbol + ": " + e.getMessage());
+        ensureNumberOfValuesIsCorrect(values.length);
+        if (importSetOfGenes.add(gene.getEntrezGeneId())) {
+            daoGeneticAlteration.addGeneticAlterations(geneticProfileId, gene.getEntrezGeneId(), values);
+            return true;
         }
+        String geneSymbolMessage = "";
+        if (geneSymbol != null && !geneSymbol.equalsIgnoreCase(gene.getHugoGeneSymbolAllCaps())) {
+            geneSymbolMessage = " (given as alias in your file as: " + geneSymbol + ")";
+        }
+        ProgressMonitor.logWarning(format(
+            "Gene %s (%d)%s found to be duplicated in your file. Duplicated row will be ignored!",
+            gene.getHugoGeneSymbolAllCaps(),
+            gene.getEntrezGeneId(),
+            geneSymbolMessage)
+        );
+        return false;
+    }
+
+
+    /**
+     * Universal method that stores values for different genetic entities
+     * @param geneticEntityId
+     * @param values
+     * @return true if entity has been stored, false - if entity already existed
+     * @throws DaoException
+     */
+    public boolean store(
```

Comment:
This is interesting. Either we did not previously support the import of genetic entity profiles, or this functionality was provided elsewhere (perhaps it is moving here now)

Reply:
code to import genetic entity profiles was repeated in multiple importers. I moved it here.